### PR TITLE
Hook called after master received signal

### PR DESCRIPTION
**Pull request by MrKiven**

### CODE REVIEW

File: gunicorn/arbiter.py [hunk 1]
```diff
@@ -192,6 +192,7 @@ def run(self):
                 if not handler:
                     self.log.error("Unhandled signal: %s", signame)
                     continue
+                self.cfg.on_signal(sig)
```
Comment (wooparadog) [inline gunicorn/arbiter.py]: self.cfg.on_signal(self, sig)

File: gunicorn/config.py [hunk 2]
```diff
@@ -1600,6 +1600,22 @@ def on_exit(server):
...
 
+class OnSignal(Setting):
+    name = "on_signal"
+    section = "Server Hooks"
+    validator = validate_callable(2)
+
+    def on_signal(self, sig):
+        pass
+
+    default = classmethod(on_signal)
```
Comment (wooparadog) [inline gunicorn/config.py]: should be staticmethod

`def on_signal` should be `def on_signal(arbitor, signal)`

**Comment by MrKiven** [comment]
Done

File: gunicorn/config.py [hunk 3]
```diff
@@ -1603,12 +1603,12 @@ def on_exit(server):
 class OnSignal(Setting):
     name = "on_signal"
     section = "Server Hooks"
-    validator = validate_callable(2)
+    validator = validate_callable(-1)
```
Comment (wooparadog) [inline gunicorn/config.py]: 呃…为啥是-1

Comment (MrKiven) [inline gunicorn/config.py]: -1.. 不检测钩子函数参数长度..可以适用于实例方法..

**Pull request merged**